### PR TITLE
PR: Add an option to run a file in the console's current namespace

### LINKED ISSUE #4180: It should be possible to get the same behaviour with Spyder's Run button than with the Ipython %run magic command
It should be possible to get the same behaviour with Spyder's Run button than with the Ipython %run magic command.

With a buggy script script.py with only `print(a)`, we can get with Ipython:

In [1]: a, b = 1, 2

In [2]: run script.py
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
/home/me/tmp/script.py in <module>()
      1 
----> 2 print(a)

NameError: name 'a' is not defined

In [3]: print(a, b)
1 2

The matlabish "clear all" option in Spyder is not sufficient.

Thank you for your work on Spyder.

### PR DESCRIPTION
**Pull request by bcolsen**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [x] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

This allows running a file in the current namespace of the IPython console. This is necessary to work around the new default of running files in a new namespace and appending the results into the current console. See https://github.com/spyder-ide/spyder-kernels/pull/131

image

image

### Issue(s) Resolved

Related to #4180 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: bcolsen

### CODE REVIEW

**Comment by ccordoba12** [comment]
@bcolsen, please merge with master to get the fix for CircleCI.

File: spyder/preferences/runconfig.py [hunk 1]
```diff
@@ -88,6 +90,8 @@ def set(self, options):
         self.python_args_enabled = options.get('python_args/enabled', False)
         self.clear_namespace = options.get('clear_namespace',
                                     CONF.get('run', 'clear_namespace', False))
+        self.console_namespace = options.get('console_namespace',
+                                   CONF.get('run', 'console_namespace', False))
```
Comment (bcolsen) [inline spyder/preferences/runconfig.py]: This a PEP error but fixing it doesn't make it more readable

Comment (ccordoba12) [reply spyder/preferences/runconfig.py]: Ok, don't worry about it.

**Comment by ccordoba12** [comment]
Note: The failure in Azure is unrelated to this.

**Pull request merged**